native/cocos/renderer/gfx-metal/MTLSemaphore.h

```diff
@@ -32,38 +32,52 @@ namespace gfx {
 
 class CCMTLSemaphore final {
 public:
-    explicit CCMTLSemaphore(uint32_t initialValue) : _semaphoreCount(initialValue) {
+    explicit CCMTLSemaphore(uint32_t initialValue) : _semaphoreCount(initialValue), _restCount(initialValue) {
         _semaphore = dispatch_semaphore_create(initialValue);
     }
-    ~CCMTLSemaphore() = default;
+    
+    ~CCMTLSemaphore() {
+        if(_semaphore) {
+            while(_restCount-- >0) {
```

Comment:
`_restCount ` is minus in `signal()`, duplicate?

Reply:
https://man.archlinux.org/man/dispatch_semaphore_create.3.en to ensure signal at least 3 semphores when released.